docs/development/style_guide.md

```diff
@@ -0,0 +1,583 @@
+
+[Python pathlib documentation](https://docs.python.org/3/library/pathlib.html)
+
+❌ **Bad:** String-based path manipulation
+
+```python
+import os
+
+# Hard to read, platform-specific separators
+artifact_path = output_dir + "/" + artifact_group + "/" + "rocm.tar.gz"
+
+# Verbose and error-prone
+if os.path.exists(os.path.join(base_dir, "build", "artifacts")):
+    files = os.listdir(os.path.join(base_dir, "build", "artifacts"))
+```
+
+✅ **Good:** Using `pathlib.Path`
+
+```python
+from pathlib import Path
+
+# Clear, readable, platform-independent
+artifact_path = Path(output_dir) / artifact_group / "rocm.tar.gz"
+
+# Concise and type-safe
+artifacts_dir = Path(base_dir) / "build" / "artifacts"
+if artifacts_dir.exists():
+    files = list(artifacts_dir.iterdir())
+```
+
+Benefits:
+
+- **Platform-independent:** Handles Windows vs Unix path separators, symlinks,
+  and other features automatically
+- **Readable:** `/` operator is clearer than string concatenation
+- **Type-safe:** Catches errors at development time
+- **Feature-rich:** Built-in methods like `.exists()`, `.mkdir()`, `.glob()`
+
+> [!TIP]
+> See the official
+> ["Corresponding tools" documentation](https://docs.python.org/3/library/pathlib.html#corresponding-tools)
+> for a table mapping from various `os` functions to `Path` equivalents.
+
+### Don't make assumptions about the current working directory
+
+Scripts should be runnable from the repository root, their script subdirectory,
+and other locations. They should not assume any particular current working
+directory.
+
+Benefits:
+
+- **Location-independent:** Script works from any directory
+- **Explicit:** Clear where files are relative to the script
+- **CI-friendly:** Works in CI environments with varying working directories,
+  especially when scripts and workflows are used in other repositories
+
+❌ **Bad:** Assuming script runs from a specific directory
+
+```python
+from pathlib import Path
+
+# Assumes script is run from repository root
+config_file = Path("build_tools/config.json")
+
+# Assumes script is run from its own directory
+data_file = Path("../data/artifacts.tar.gz")
+```
+
+✅ **Good:** Paths relative to the script location
+
+```python
+from pathlib import Path
+
+# Establish script's location as reference point
+THIS_SCRIPT_DIR = Path(__file__).resolve().parent
+THEROCK_DIR = THIS_SCRIPT_DIR.parent
+
+# Build paths relative to script location
+config_file = THIS_SCRIPT_DIR / "config.json"
+# Build paths relative to repository root
+version_file = THEROCK_DIR / "version.json"
+```
+
+### Use `argparse` for CLI flags
+
+Use [`argparse`](https://docs.python.org/3/library/argparse.html) for
+command-line argument parsing with clear help text and type conversion.
+
+Benefits:
+
+- **Automatic help:** Users get `-h/--help` for free
+- **Type conversion:** Arguments are converted to correct types
+- **Validation:** Required arguments are enforced
+
+❌ **Bad:** Manual argument parsing
```

Comment:
just remove the entire manual argument parsing section. just show how it is done in argparse.

for other sections the good/bad part are helpful but here i think this could only give wrong ideas. unless you come from C you dont do manual parsing anyway

Reply:
> unless you come from C you dont do manual parsing anyway

If only... I've seen this code very recently: https://github.com/ROCm/rocm-libraries/pull/2411/files#r2519049187

<img width="855" height="573" alt="Image" src="https://github.com/user-attachments/assets/ea03a747-df2a-46fb-88c1-de2ff5af6819" />

I think it's worth keeping the example so people can pattern match code "oh yeah, I saw that somewhere, let me go fix it".